File: public/actions/javascript/addToIndex.ts
import { Findex, IndexedEntry } from "cloudproof_js";

// interface IndexedEntry {
//   indexedValue: IndexedValue | Location | Keyword;
//   keywords: Set<Keyword> | Keyword[] | string[];
// }

//
// Index your data
//
export const addToIndex = async (findex: Findex, associations: IndexedEntry[]): Promise<void> => {
  await findex.add(associations);
};

File: src/actions/java/addToIndex.java
import com.cosmian.jna.findex.Findex;
//
// Index your data
//
public static Map<Keyword, Set<Location>> addToIndex(Findex findex, Map<IndexedValue, Set<Keyword>> indexedValuesAndWords) {
  return findex.add(indexedValuesAndWords)
}
